src/agent/coverage/src/block.rs

```diff
@@ -280,4 +377,58 @@ mod tests {
 
         assert_eq!(total, expected);
     }
+
+    #[test]
+    fn test_block_cov_serde() {
+        let block = BlockCov { offset: 0x123, count: 456 };
+
+        let ser = serde_json::to_string(&block).unwrap();
+
+        let text = r#"{"offset":"0x123","count":456}"#;
+        assert_eq!(ser, text);
+
+        let de: BlockCov = serde_json::from_str(&ser).unwrap();
+        assert_eq!(de, block);
+    }
+
+    #[test]
+    fn test_cmd_cov_serde() {
+        #[cfg(target_os = "linux")]
+        const MAIN_EXE: &str = "/main.exe";
+
+        #[cfg(target_os = "linux")]
+        const SOME_DLL: &str = "/lib/some.dll";
+
+        #[cfg(target_os = "windows")]
+        const MAIN_EXE: &str = r"c:\main.exe";
+
+        #[cfg(target_os = "windows")]
+        const SOME_DLL: &str = r"c:\lib\some.dll";
+
+        let main_exe = module_path(MAIN_EXE);
+        let some_dll = module_path(SOME_DLL);
+
+        let cov = {
+            let mut cov = CommandBlockCov::default();
+            cov.insert(&main_exe, vec![0x1, 0x20, 0x300].into_iter());
+            cov.increment(&main_exe, 0x1);
+            cov.increment(&main_exe, 0x300);
+            cov.insert(&some_dll, vec![0x2, 0x30, 0x400].into_iter());
+            cov.increment(&some_dll, 0x30);
+            cov
+        };
+
+        let ser = serde_json::to_string(&cov).unwrap();
+
+        #[cfg(target_os = "linux")]
+        let text = r#"{"/lib/some.dll":[{"offset":"0x2","count":0},{"offset":"0x30","count":1},{"offset":"0x400","count":0}],"/main.exe":[{"offset":"0x1","count":1},{"offset":"0x20","count":0},{"offset":"0x300","count":1}]}"#;
+
+        #[cfg(target_os = "windows")]
```

Comment:
This is bad, as is the conditional compilation in the existing tests. This has to do with how `ModulePath` validates absolute paths in an OS-dependent way. Will address separately.